Marlin/src/gcode/calibrate/G34_M422.cpp

```diff
@@ -250,6 +284,7 @@ void GcodeSuite::G34() {
       bltouch._stow();
     #endif
 
+    // Home after the alignment procedure
     gcode.G28(false);
```

Comment:
If parameters 'O', 'R', 'S', 'X', 'Y', or 'Z' are passed to `G34` then they will be picked up by `G28` here. The same problem will exist with code that calls `home_all_axes`. So, instead of homing this way, we should make `USE_EXECUTE_COMMANDS_IMMEDIATE` always available and call `process_subcommands_now_P`, which prevents parameter conflicts.

Reply:
I've patched this to use the G-code macro call. Another approach would have been to simply reset the parser ahead of the call to `gcode.G28`, but that still left `home_all_axes` unsafe.